src/hooks/useCreateEvmDonation.tsx

```diff
@@ -27,6 +27,7 @@ export const useCreateEvmDonation = () => {
 	const { chain } = useNetwork();
 	const chainId = chain?.id;
 	const isSafeEnv = useIsSafeEnvironment();
+	const draftDonationIdRef = useRef(0);
```

Comment:
Would it cause any issue if we save `draftDonationId` as part of `createDonationProps`? if no, it will make the code cleaner and needs less modification